src/python/gem5/components/processors/simpoint.py

```diff
@@ -0,0 +1,99 @@
+
+
+from m5.util import *
+from pathlib import Path
+from typing import List, Tuple
+
+
+class Simpoint:
+    """
+    This Simpoint class is used to manage the information needed for simpoint
+    in workload
+
+    """
+
+    def __init__(
+        self,
+        simpoint_interval = None,
+        simpoint_file_path: Path = None,
+        weight_file_path: Path = None,
+        simpoint_list: List[int] = None,
+        weight_list: List[int] = None,
+        # bbv_file_path: Path = None, will take in bbv and generate simpoints
+        # with it in the future
+    ) -> None:
+        """
+        :param simpoint_interval: the length of each simpoint interval
+        :param simpoint_file_path: the path to the simpoint result file
+        generated by Simpoint3.2 or gem5
+        :param weight_file_path: the path to the weight result file generated
+        by Simpoint3.2 or gem5
+
+        :param simpoint_list: a list of simpoint starting instructions (sorted)
```

Comment:
Sorted how? What us the sorting?

Reply:
The simpoint_list is a list of the simpoint interval numbers (ex. [3, 5, 15]). Sorry about the names, I'm bad at naming. They should be sorted in ascending order. This might be good for cases that need to schedule starting instructions one by one during the simulation. But now I think about it, you're right, they don't need to be sorted.

Reply:
I see what you're saying now. Sorted based on the instruction number, not based on the weights or anything else.